include/swift/Option/Options.td

```diff
@@ -554,6 +554,10 @@ def dump_parse : Flag<["-"], "dump-parse">,
 def dump_ast : Flag<["-"], "dump-ast">,
   HelpText<"Parse and type-check input file(s) and dump AST(s)">, ModeOpt,
   Flags<[FrontendOption, NoInteractiveOption, DoesNotAffectIncrementalBuild]>;
+def dump_serialized_syntax_tree : Flag<["-"], "dump-serialized-syntax-tree">,
+  HelpText<"Parse input file(s) and dump the Syntax tree(s) as JSON">, ModeOpt,
+  Flags<[FrontendOption, NoInteractiveOption, HelpHidden,
+         DoesNotAffectIncrementalBuild]>;
```

Comment:
This makes it not just a frontend option but also a driver option. If you want it to just be a frontend option it would go in FrontendOptions.td.